File: Sakila_Net/Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDInventory.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOInventory 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  inventory_id ,film_id ,store_id ,last_update  from inventory";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClInventory inventory)  
        {
            String stSql = "Select  inventory_id ,film_id ,store_id ,last_update  from inventory";
            stSql += " Where  inventory_id='" + inventory.InventoryId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        inventory.FilmId = new ClFilm(int.Parse(dt.Rows[0]["film_id"].ToString()));
        inventory.StoreId = new ClStore(int.Parse(dt.Rows[0]["store_id"].ToString()));
        inventory.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClInventory inventory)  
        {
            String stSql = "insert into Inventory (" 
                       + " inventory_id ,film_id ,store_id ,last_update" 
                           + " )values (" 
            + "'" + inventory.InventoryId + "'" 
                     + ",'" + inventory.FilmId.FilmId + "'" 
                     + ",'" + inventory.StoreId.StoreId + "'" 
                     + ",'" + inventory.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClInventory inventory) {
            
            String stSql  =  "delete from Inventory " 
             + " Where  inventory_id='" + inventory.InventoryId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClInventory inventory ) {
            String stSql  =  "update Inventory SET " 
             + " film_id = '" + inventory.FilmId.FilmId + "'" 
                             + ",store_id = '" + inventory.StoreId.StoreId + "'" 
                             + ",last_update = '" + inventory.LastUpdate + "'" 
             + " Where  inventory_id='" + inventory.InventoryId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}

File: Sakila_Net/Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDInventory.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOInventory
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from inventory"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( inventory as ClInventory) as Boolean
             Dim StSql as String = "Select * from inventory"
            StSql &= " Where  inventory_id='" & inventory.dataInventoryId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        inventory.DataFilmId = new ClFilm(Integer.Parse(dt.Rows(0)("film_id")))
        inventory.DataStoreId = new ClStore(Integer.Parse(dt.Rows(0)("store_id")))
        inventory.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( inventory as ClInventory)  as Integer
            Dim stSql as String = "insert into Inventory (" _
                       & " inventory_id ,film_id ,store_id ,last_update" _
                           & " )values (" _
            & "'" & inventory.dataInventoryId & "'" _
                     & ",'" & inventory.dataFilmId.datafilmId & "'" _
                     & ",'" & inventory.dataStoreId.datastoreId & "'" _
                     & ",'" & inventory.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( inventory as ClInventory) as Integer
            
            Dim stSql as String =  "delete from Inventory " _
             & " Where  inventory_id='" & inventory.dataInventoryId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(inventory as ClInventory) as Integer
            Dim stSql as String =  "update Inventory SET " _
             & " film_id = '" & inventory.DataFilmId.DataFilmId & "'" _
                             & ",store_id = '" & inventory.DataStoreId.DataStoreId & "'" _
                             & ",last_update = '" & inventory.DataLastUpdate & "'" _
             & " Where  inventory_id='" & inventory.dataInventoryId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  
